main.tf

```diff
@@ -71,6 +71,82 @@ resource "aws_iam_user_policy" "quortex_bucket_rw" {
 EOF
 }
 
+# --- Roles and policies associated to irsa ---
+data "aws_caller_identity" "current" {}
+
+resource "aws_iam_role" "aws_eks_irsa" {
+  for_each           = local.buckets
+  name               = "${var.storage_prefix}-${each.key}-irsa"
+  assume_role_policy = data.aws_iam_policy_document.irsa_assume_role_policy[each.key].json
+}
```

Comment:
The IRSA role should be behind a switch in the variables, and the aws_iam_user.quortex, aws_iam_access_key.quortex, and the user aws_iam_user_policy.quortex_bucket_rw should also be behind a similar switch.

Reply:
You forgot adding a switch to make creation of user / access keys optional.